lib/spack/spack/solver/asp.py

```diff
@@ -1522,6 +1522,7 @@ def external_spec_selected(self, pkg, idx):
         self._specs[pkg].external_modules = (
             spack.spec.Spec._format_module_list(spec_info.get('modules', None))
         )
+        spack.spec.Spec.ensure_external_path_if_external(self._specs[pkg])
```

Comment:
Why do we need this call here? Isn't:

https://github.com/spack/spack/blob/372fc78e9869e36c54ac86bfd20e2447df0a8405/lib/spack/spack/solver/asp.py#L1636-L1638 

sufficient to ensure `external_path` is set?

Reply:
That will probably need to be a bigger fix though